.github/workflows/ci.yml

```diff
@@ -27,6 +27,9 @@ jobs:
         with:
           version: ${{ matrix.version }}
           arch: ${{ matrix.arch }}
+      - run: julia --project -e 'using Pkg; Pkg.develop([PackageSpec(path="SnoopCompileCore"),
```

Comment:
Is this the correct thing to do? @timholy When someone adds `SnoopCompile`, these packages are not developed by default.

Reply:
When they add it, it will be registered. In principle, this is a good workaround for https://github.com/JuliaLang/Pkg.jl/issues/1874 (better than committing the Manifest).

Reply:
Because this is similar to https://github.com/timholy/SnoopCompile.jl/commit/bb56316297de70d401e84433bf84064cd9cee9fe, I am confused. If developing is a bad thing we should not do it at all (including in CI). If it is a good thing when we don't do it by default (when someone adds the package).

Reply:
> If running this script is a bad thing we should not do it at all (including in CI).

For test usage, it'll need a `dev`ed version of these sub-packages to make sure we can submit three releases altogether. While on the other hand, common users should just install the released version of these subpackages as dependencies.

`Pkg.develop` in `deps/build.jl` is like pulling arbitrary code from the internet and install it; there's no quality and version control here.

Reply:
That means people cannot develop SnoopCompile directly, and we might want to add this to the contribution documentation.